client/landing/domains/registrant-verification/index.jsx

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DomainsLandingHeader from '../header';
+import DomainsLandingContentCard from '../content-card';
+import { CALYPSO_CONTACT } from 'lib/url/support';
+import wp from 'lib/wp';
+
+const wpcom = wp.undocumented();
+
+class RegistrantVerificationPage extends Component {
+	static propTypes = {
+		domain: PropTypes.string.isRequired,
+		email: PropTypes.string.isRequired,
+		token: PropTypes.string.isRequired,
+	};
+
+	state = {
+		isLoading: true,
+		success: false,
+		error: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = this.getLoadingState();
+	}
+
+	componentWillMount() {
+		const { domain, email, token } = this.props;
+		wpcom.domainsVerifyRegistrantEmail( domain, email, token ).then(
+			() => {
+				this.setState( this.getVerificationSuccessState() );
+			},
+			error => {
+				this.setErrorState( error );
+			}
+		);
+	}
+
+	getLoadingState = () => {
+		const { translate } = this.props;
+		return {
+			isLoading: true,
+			illustration: '/calypso/images/illustrations/whoops.svg',
+			title: translate( 'Verifying your contact information…' ),
+			message: 'Loading…',
+			actionTitle: null,
+			actionCallback: null,
+			footer: 'Loading…',
+		};
+	};
+
+	getVerificationSuccessState = () => {
+		const { domain, translate } = this.props;
+		return {
+			illustration: '/calypso/images/illustrations/illustration-ok.svg',
+			title: translate( 'Success!' ),
+			message: translate(
+				'Thank your for verifying your contact information for:{{br /}}{{strong}}%(domain)s{{/strong}}.',
```

Comment:
The `:` seems a bit clunky (as if we supported verifying more than one domain at once? How about dropping it and making the `strong` `white-space: nowrap` instead?